chrome/content/zotero/xpcom/style.js

```diff
@@ -190,7 +190,26 @@ Zotero.Styles = new function() {
 		}
 		
 		return _styles[id] || false;
-	}
+	};
+	
+	this.getAsync = Zotero.Promise.coroutine(function* (id, skipMappings, download=true) {
+		yield this.init();
+		var style = this.get(id, skipMappings);
+		
+		if (!style && download) {
+			try {
+				// Don't wait too long
+				Zotero.debug(`Fetching required style ${id} asynchronously`);
+				let styleText = yield Zotero.File.getContentsFromURLAsync(id, {timeout: 5000});
```

Comment:
This is not entirely on-point either, because we're getting the id from the document, where that's the only thing we store. Although this fact is a decent case to actually store rel="self" in the document prefs as well for future compatibility.